File: testdata/ExtractAndMoveMethod/Cpp/parent/OrderProcessor.cpp
#include <iostream>
#include <vector>
#include <string>

class PriceCalculator {

public:
    double getItemPrice(const std::string& item) {
        // Imagine this fetches price from a database
        return 20.0;
    }
};

class OrderProcessor {

public:
    void processOrder(const std::vector<std::string>& items) {
        double total = 0;
        for (const auto& item : items) {
            total += priceCalculator.getItemPrice(item);
        }
        applyDiscount(total);
        std::cout << "Total: " << total << std::endl;
    }

private:
    PriceCalculator priceCalculator;

    void applyDiscount(double total) {
        // Apply a discount if total is above a certain amount
        if (total > 100) {
            total *= 0.9;
        }
    }
};

File: testdata/ExtractAndMoveMethod/Java/current/OrderProcessor.java
import java.util.List;

public class OrderProcessor {

    private PriceCalculator priceCalculator = new PriceCalculator();

    public void processOrder(List<String> items) {
        double total = priceCalculator.calculateTotal(items);
        applyDiscount(total);
        System.out.println("Total: " + total);
    }

    private void applyDiscount(double total) {
        // Apply a discount if total is above a certain amount
        if (total > 100) {
            total *= 0.9;
        }
    }
}

public class PriceCalculator {

    public double calculateTotal(List<String> items) {
        double total = 0;
        for (String item : items) {
            total += getItemPrice(item);
        }
        return total;
    }

    public double getItemPrice(String item) {
        // Imagine this fetches price from a database
        return 20.0;
    }
}
